components/page-stack/hooks.ts

```diff
@@ -1,6 +1,32 @@
-import { useContext } from "../../deps.ts";
-import { PageStackContext } from './context.ts'
+import { signal } from "../../deps.ts";
+
+export interface PageState {
+  Component: React.ReactNode;
+  isEscapeDisabled: boolean;
+}
+
+const pageStack$ = signal<PageState[]>([]);
```

Comment:
this is nice

Reply:
yeah got rid of a ton of code to do the same thing 😅